services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java

```diff
@@ -560,15 +583,17 @@ public Response addExecutionToFlow(@Parameter(description = "Alias of parent flo
                 // useful for internal providers that already provide a built-in configuration
                 AuthenticatorConfigRepresentation configRepresentation = ModelToRepresentation.toRepresentation(
                         config);
-                newExecutionConfig(execution.getId(), configRepresentation).close();
+
+                if (config.getId() != null && realm.getAuthenticatorConfigById(config.getId()) != null) {
```

Comment:
And same here? Just reuse config if READ_ONLY_CONFIGURATION? I don't know if there are cases when there is a default config, but it can be modified to have two different configs later...